files/common/usr/bin/unpack-image

```diff
@@ -96,22 +117,35 @@ tar -xzf "$UPGRADE_IMAGE_PATH" ||
 
 report_progress_inc 40 "Verifying format."
 
+# check if this is new style upgrade image
+if [[ -f "complete-unpack" ]]; then
+
+	# verify signatures and hashes before hand-off
+	verify_sig $opt_s
+
+	#
+	# This is newstyle upgrade image that brings with it logic to complete
+	# the unpack process. Hand-off the rest of unpack steps to the script within.
+	#
+	report_progress_inc 50 "Handoff unpack to new script."
+
+	"$(pwd)"/complete-unpack ${f_str:+"-f"} ${s_str:+"-s"} ${x_str:+"-x"} ||
```

Comment:
I don't really understand why we use these "_str" variables, vs. just passing in the original value of "$@" passed into this script (we might need to copy it first, since I don't know if `getopts` will mangle it above).

Futher, if we want to use the `${:+}` construct, why not use the "opt_" variables? Why do we need these "_str" variables?

E.g. I'd most prefer something like:
```
"$(pwd)"/complete-unpack "$@"
```
But if we can't do that, perhaps we could do:
```
"$(pwd)"/complete-unpack ${opt_f:+"-f"} ...
```
?

Also, can we drop the use of `"$(pwd)"` here.. I don't see why it's required.. can't we more simply do: `./complete-unpack` ?

Reply:
So the behavior of the `${:+}` construct seems to require a null value in order for the replacement to not occur. In the script as is, opt_f are all not-null already - with a value of false. My original implementation had me recording the input parameters in a string and passing that in. But this failed the shell formatting check which asked the string variable to be enclosed with double quotes. And enclosing in double quotes makes it a single long input rather than multiple input options.

using `$@` I thought was not available since we use shift. Let me take a look to see if it works.

Reply:
Yea, I see.. so maybe your implementation is actually the best we can do, given the constraints.. I think Bash is lacking here, and making this a little more convoluted than it might be in another language..

If we keep your implementation, do we still need to initialize the `_str` variable to the empty string? I'd be OK dropping the initialization if we can avoid it.

Reply:
ya the initialization seems redundant. can remove those. Will wait for us to decide on whether to create the new unpack-image/prepare script in appliance-build repo before addressing the code flow comments.